hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/NoOpDataBlockEncoder.java

```diff
@@ -47,12 +47,14 @@ private NoOpDataBlockEncoder() {
   }
 
   @Override
-  public int encode(Cell cell, HFileBlockEncodingContext encodingCtx,
+  public void encode(Cell cell, HFileBlockEncodingContext encodingCtx,
       DataOutputStream out) throws IOException {
     NoneEncodingState state = (NoneEncodingState) encodingCtx
         .getEncodingState();
+    int posBeforeEncode = out.size();
     NoneEncoder encoder = state.encoder;
-    return encoder.write(cell);
+    int size = encoder.write(cell);
+    state.bookKeepingPostCellEncode(size, out.size() - posBeforeEncode);
```

Comment:
You can pass size only here for both the encoded and unEncoded correct? Seems we need that value for block boundary cases.

Reply:
Done